FirebaseDatabase/Sources/Api/FIRDatabase.m

```diff
@@ -173,6 +173,17 @@ - (void)purgeOutstandingWrites {
     });
 }
 
+- (void)useEmulatorWithHost:(NSString *)host port:(NSInteger)port {
+    NSAssert(host.length > 0, @"Cannot connect to nil or empty host");
+    NSAssert(self.repo == nil,
+             @"Cannot connect to emulator after database initialization. Call "
+             @"useEmulator(host:port:) before creating a database reference or "
+             @"opening a database connection.");
+    NSString *fullHost =
+        [NSString stringWithFormat:@"%@:%li", host, (long)port];
+    self.repoInfo.host = fullHost;
+}
+
```

Comment:
Can you also port Sam's change to https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseDatabase/Sources/Api/FIRDatabase.m#L156? We decided to only validate the host in `referenceFromUrl` if the underlying URL does not point to a custom host. This would allow a user to use production URLs in `referenceFromUrl` even if the use an emulator.

Reply:
Done.